pkg/gatherers/clusterconfig/gather_aggregated_instances_test.go

```diff
@@ -0,0 +1,183 @@
+		expected  []string
+	}{
+		{
+			name: "The function returns the name of the Prometheus outside the 'openshift-monitoring' namespace",
+			alertMgrs: []*v1.Alertmanager{
+				{ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "test-namespace"}},
+			},
+			expected: []string{"test"},
+		}, {
+			name: "The function ignores the name of the Prometheus inside the 'openshift-monitoring' namespace",
+			alertMgrs: []*v1.Alertmanager{
+				{ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "openshift-monitoring"}},
+			},
+			expected: []string{},
+		}, {
+			name: "The function returns only items outside of the namespace on a mixed response from client",
+			alertMgrs: []*v1.Alertmanager{
+				{ObjectMeta: metav1.ObjectMeta{Name: "test1", Namespace: "test-namespace"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "ignore", Namespace: "openshift-monitoring"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "test2", Namespace: "test-namespace"}},
+			},
+			expected: []string{"test1", "test2"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Given
+			items := make([]runtime.Object, len(tc.alertMgrs))
+			for i := range tc.alertMgrs {
+				items[i] = tc.alertMgrs[i]
+			}
+			clientset := fake.NewSimpleClientset(items...)
+
+			// When
+			test, err := aggregatedInstances{}.getOutcastedAlertManagers(context.Background(), clientset)
+
+			// Assert
+			assert.NoError(t, err)
+			assert.EqualValues(t, tc.expected, test)
+		})
+	}
+}
+
+// Test_getOutcastedPrometheuses provides unit tests for the namespace filtering logic of Prometheus instances
+func Test_getOutcastedPrometheuses(t *testing.T) {
+	testCases := []struct {
+		name     string
+		proms    []*v1.Prometheus
+		expected []string
+	}{
+		{
+			name: "The function returns the name of the Prometheus outside the 'openshift-monitoring' namespace",
+			proms: []*v1.Prometheus{
+				{ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "test-namespace"}},
+			},
+			expected: []string{"test"},
+		}, {
+			name: "The function ignores the name of the Prometheus inside the 'openshift-monitoring' namespace",
+			proms: []*v1.Prometheus{
+				{ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "openshift-monitoring"}},
+			},
+			expected: []string{},
+		}, {
+			name: "The function returns only items outside of the namespace on a mixed response from client",
+			proms: []*v1.Prometheus{
+				{ObjectMeta: metav1.ObjectMeta{Name: "test1", Namespace: "test-namespace"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "ignore", Namespace: "openshift-monitoring"}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "test2", Namespace: "test-namespace"}},
+			},
+			expected: []string{"test1", "test2"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Given
+			items := make([]runtime.Object, len(tc.proms))
+			for i := range tc.proms {
+				items[i] = tc.proms[i]
+			}
+			clientset := fake.NewSimpleClientset(items...)
```

Comment:
I just found out that you can optionally do something like:
```
			clientset := fake.NewSimpleClientset()
			for _, p := range tc.proms {
				clientset.Tracker().Add(p)
			}
			for _, a := range tc.alertMgrs {
				clientset.Tracker().Add(a)
			}
```

Reply:
I liked the idea, thanks. The other workaround felt messy.